tests/models/bark/test_modeling_bark.py

```diff
@@ -0,0 +1,2486 @@
+                1002,
+                1002,
+                1002,
+                1002,
+            ],
+            [
+                806,
+                685,
+                905,
+                848,
+                803,
+                810,
+                921,
+                208,
+                625,
+                203,
+                595,
+                850,
+                840,
+                499,
+                653,
+                999,
+                254,
+                361,
+                996,
+                901,
+                433,
+                803,
+                602,
+                982,
+                504,
+                214,
+                229,
+                718,
+                18,
+                371,
+                534,
+                534,
+                1012,
+                534,
+                480,
+                526,
+                416,
+                458,
+                624,
+                937,
+                1015,
+                495,
+                1008,
+                838,
+                542,
+                506,
+                181,
+                110,
+                208,
+                821,
+                437,
+                926,
+                939,
+                443,
+                848,
+                780,
+                746,
+                810,
+                625,
+                966,
+                658,
+                677,
+                919,
+                468,
+                832,
+                534,
+                534,
+                701,
+                835,
+                534,
+                948,
+                948,
+                948,
+                948,
+                989,
+                989,
+                989,
+                989,
+                975,
+                518,
+                475,
+                975,
+                948,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+                475,
+            ],
+        ]
+
+        output_ids = self.model.generate_text_semantic(**input_ids, history_prompt=history_prompt, do_sample=False)
+
+        output_ids = self.model.generate_coarse(
+            output_ids, history_prompt=history_prompt, max_coarse_history=630, sliding_window_len=60, do_sample=False
+        )
+
+        # greedy decoding
+        output_ids = self.model.generate_fine(
+            output_ids,
+            history_prompt=history_prompt,
+            temperature=None,
+        )
+
+        self.assertListEqual(output_ids[0, :, :100].tolist(), first_hundred_expected_output_ids)
+
+    @slow
+    def test_generate(self):
```

Comment:
This is going to be quite slow generating with the full model six times - can we split this into maybe two separate tests that do just two of the six test cases?